src/main/java/isaac/bastion/listeners/ModeListener.java

```diff
@@ -0,0 +1,155 @@
+package isaac.bastion.listeners;
+
+import isaac.bastion.Bastion;
+import isaac.bastion.BastionBlock;
+import isaac.bastion.utils.BastionSettingManager;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerMoveEvent;
+import vg.civcraft.mc.civmodcore.playersettings.PlayerSetting;
+import vg.civcraft.mc.civmodcore.playersettings.SettingChangeListener;
+import vg.civcraft.mc.civmodcore.playersettings.impl.DisplayLocationSetting;
+import vg.civcraft.mc.civmodcore.scoreboard.bottom.BottomLine;
+import vg.civcraft.mc.civmodcore.scoreboard.bottom.BottomLineAPI;
+import vg.civcraft.mc.civmodcore.scoreboard.side.CivScoreBoard;
+import vg.civcraft.mc.civmodcore.scoreboard.side.ScoreBoardAPI;
+
+import java.util.Set;
+import java.util.UUID;
+
+
+public class ModeListener implements Listener {
+
+	private BottomLine bsiBottomLine;
+	private CivScoreBoard bsiBoard;
+	private BastionSettingManager settingMan;
+
+	public ModeListener() {
+		this.bsiBoard = ScoreBoardAPI.createBoard("bsiDisplay");
+		this.bsiBottomLine = BottomLineAPI.createBottomLine("bsiDisplay", 3);
+		this.settingMan = Bastion.getSettingManager();
+		settingMan.getBsiOverlay().registerListener(new SettingChangeListener<Boolean>() {
+			@Override
+			public void handle(UUID player, PlayerSetting<Boolean> setting, Boolean oldValue, Boolean newValue) {
+				setBsiOverlay(Bukkit.getPlayer(player), newValue);
```

Comment:
Just call `checkLocationForBastions()` here and in the listener below and remove the method `setBsiOverlay` entirely

Reply:
When doing this and disabling the setting in /config, it will just stick with whatever was currently in that line instead of hiding it

Reply:
Then do
```
if (newValue) {
    checkLocationForBastions(..);
}
else {
		bsiBoard.hide(player);
			bsiBottomLine.removePlayer(player);
}